sign.mjs

```diff
@@ -19,21 +31,28 @@ const siweMessage = new siwe.SiweMessage({
   uri: origin,
   version: "1",
   chainId: "1",
+  nonce,
+  expirationTime,
 });
 
 const messageToSign = siweMessage.prepareMessage();
 
 const signature = await wallet.signMessage(messageToSign);
 
-console.log("signature", signature);
-
 const recoveredAddress = ethers.utils.verifyMessage(messageToSign, signature);
 
+if (recoveredAddress !== wallet.address) {
+  throw new Error("Recovered address does not match wallet address");
+}
+
+const latestBlockhash = litNodeClient.getLatestBlockhash();
+console.log("latestBlockhash", latestBlockhash);
+
 const authSig = {
   sig: signature,
   derivedVia: "web3.eth.personal.sign",
   signedMessage: messageToSign,
-  address: recoveredAddress,
+  address: await wallet.getAddress(),
```

Comment:
nit: for the sake of the example we shouldnt inline the `await` since if it fails for some reason the stack trace will be pretty dense.

Reply:
ah good catch!